simpleble/src/backends/macos/AdapterBaseMacOS.mm

```diff
@@ -101,6 +101,10 @@ - (void)centralManager:(CBCentralManager*)central
     // Extract RSSI
     advertisingData.rssi = [RSSI shortValue];
 
+    // Extract Tx Power
+    NSNumber* txPower = advertisementData[CBAdvertisementDataTxPowerLevelKey];
```

Comment:
Similar question as with Linux: How are default values expressed here if the data is not present? I'd suggest following the same approach of using an uninitialized default value which gets overwritten if advertising data is included.